src/bun.js/api/server.zig

```diff
@@ -2961,6 +2966,75 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
             this.finishRunningErrorHandler(value, status);
         }
 
+        fn processOnErrorPromise(
+            ctx: *RequestContext,
+            promise_js: JSC.JSValue,
+            promise: JSC.AnyPromise,
+            value: JSC.JSValue,
+            status: u16,
+        ) void {
+            var vm = ctx.server.vm;
+
+            switch (promise.status(vm.global.vm())) {
```

Comment:
This is a copy of what `RequestContext.onResponse` is doing. I'm not sure if this is the correct way of handling the promise.